Tests/SCM.SwissArmyKnife.Test/GzipTests/GzipTests.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.IO;
+using System.Text;
+using FluentAssertions;
+using SCM.SwissArmyKnife.GzipOperations;
+using Xunit;
+
+namespace SCM.SwissArmyKnife.Test.GzipTests
+{
+    public class GzipTests
+    {
+        private readonly string testString = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non nunc quis elit commodo placerat. Curabitur iaculis justo enim, " +
+            "sit amet scelerisque velit auctor non. Donec eleifend tortor non sem ultrices rhoncus. Mauris porttitor, erat ac faucibus interdum, risus eros rutrum sapien, sed" +
+            " lobortis sem nisi eget orci. Morbi non ex erat. Suspendisse porttitor suscipit erat, at auctor lacus placerat quis. Sed porttitor pellentesque vulputate. Nulla dictum " +
+            "eros non ligula varius porttitor. Maecenas iaculis ex non sem posuere dignissim. Cras quis nisl in erat convallis egestas. Curabitur vel mi cursus, aliquam nulla eget, " +
+            "cursus velit. Morbi sapien ipsum, malesuada eget nulla molestie, dapibus pulvinar turpis. Nam non arcu quis lorem dapibus fringilla. Nullam convallis in turpis vitae vehicula." +
+            " Aliquam erat volutpat. Nam sapien leo, gravida eu nulla in, mollis volutpat libero.";
+
+        [Fact]
+        public void Compress_IfInputThenOutput()
```

Comment:
I think we can do a little better here in regards to the tests. It's awesome you're dilligent but I think there are a few tests here that are superfluous.
For example this test would be covered by any test that also tests that we can convert the `testString` back and forth (because obviously if we can decompress it, it can't be empty)